tests/metal/groupshared-threadlocal-same-parameter.slang

```diff
@@ -0,0 +1,53 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=BUF): -shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=BUF): -vk -shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=BUF): -vk -emit-spirv-directly -shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=BUF):-slang -shaderobj -mtl
+
+//TEST_INPUT:ubuffer(data=[0 0], stride=4):out,name outputBuffer
+RWStructuredBuffer<uint> outputBuffer;
+
+static groupshared uint g_values[2] = { 1, 0 };
+static uint g_altValues[2] = { 2, 3 };
+
+static groupshared uint g_valuesReturned[2];
+static uint g_altValuesReturned[2];
+
+uint[2] maybeGroupSharedReturn(uint id)
+{
+    if (id == 0)
+    {
+        return g_values;
+    }
+    else
+    {
+        return g_altValues;
+    }
+}
+
+[numthreads(2, 1, 1)]
+void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
+{
+    AllMemoryBarrierWithGroupSync();
+    uint tid = dispatchThreadID.x;
+    if (tid == 0)
+    {
+        g_valuesReturned[tid] = maybeGroupSharedReturn(tid)[tid];
```

Comment:
This new test is failing,
```
metal 32023.35: /tmp/unknown-Hhojjc.metal(47): error :  assigning 'array<uint, int(2)> *' (aka 'array<unsigned int, int(2)> *') to 'threadgroup array<uint, int(2)> *' (aka 'threadgroup array<unsigned int, int(2)> *') changes address space of pointer
    (&kernelContext_1)->g_valuesReturned_0 = &g_valuesReturned_1;
                                             ^~~~~~~~~~~~~~~~~~~
metal 32023.35: /tmp/unknown-Hhojjc.metal(49): error :  assigning 'array<uint, int(2)> *' (aka 'array<unsigned int, int(2)> *') to 'threadgroup array<uint, int(2)> *' (aka 'threadgroup array<unsigned int, int(2)> *') changes address space of pointer
    (&kernelContext_1)->g_values_0 = &g_values_1;
                                     ^~~~~~~~~~~
2 errors generated.
```
The variables in KernelContext uses "threadgroup" whereas the output from the funciton uses "thread".

Reply:
👍 This is a bug in address space legalization that also needs to be fixed